packages/suite/src/views/settings/general/BitcoinAmountUnit.tsx

```diff
@@ -1,18 +1,30 @@
 import React from 'react';
-import { ActionColumn, ActionSelect, SectionItem, TextColumn } from '@suite-components/Settings';
+import { analytics, EventType } from '@trezor/suite-analytics';
 import { PROTO } from '@trezor/connect';
+
+import { ActionColumn, ActionSelect, SectionItem, TextColumn } from '@suite-components/Settings';
 import { Translation } from '@suite-components/Translation';
 import { useAnchor } from '@suite-hooks/useAnchor';
 import { SettingsAnchor } from '@suite-constants/anchors';
-import { useBitcoinAmountUnit } from '@wallet-hooks/useBitcoinAmountUnit';
+import {
+    BITCOIN_UNIT_ABBREVIATION,
+    useBitcoinAmountUnit,
+} from '@wallet-hooks/useBitcoinAmountUnit';
 
 export const BitcoinAmountUnit = () => {
     const { bitcoinAmountUnit, setBitcoinAmountUnits, UNIT_LABELS, UNIT_OPTIONS } =
         useBitcoinAmountUnit();
     const { anchorRef, shouldHighlight } = useAnchor(SettingsAnchor.BitcoinAmountUnit);
 
-    const handleUnitsChange = ({ value }: { value: PROTO.AmountUnit }) =>
+    const handleUnitsChange = ({ value }: { value: PROTO.AmountUnit }) => {
         setBitcoinAmountUnits(value);
+        analytics.report({
+            type: EventType.SettingsGeneralChangeBitcoinUnit,
+            payload: {
+                unit: BITCOIN_UNIT_ABBREVIATION[value],
```

Comment:
And what about doing that conversion on the analytics' side?

Reply:
Do you mean to send numbers 0, 1, 2, 3? I would like to have it human readable for QA and it is risky to add this logic to analytics scripts. It can change

Reply:
I probably meant making data human readable in the analytics package, based on the type you could format the payload. This pattern could also be used for any other places where you require conversion to human-redable data. Although now I think it might be too much of a hassle 😅